src/main/java/org/commcare/cases/instance/StorageInstanceTreeElement.java

```diff
@@ -268,8 +269,16 @@ public String intern(String s) {
     protected abstract T buildElement(StorageInstanceTreeElement<Model, T> storageInstance,
                                       int recordId, String id, int mult);
 
-    protected Model getElement(int recordId) {
-        return storage.read(recordId);
+    protected Model getElement(int recordId, QueryContext context) {
+        EvaluationTrace trace = new EvaluationTrace("Model Load[" + childName+"]");
+
+        Model m = storage.read(recordId);
+
+        trace.setOutcome(String.valueOf(recordId));
```

Comment:
Exactly. These traces are used to denote that a model was loaded directly from the DB for performance profiling (IE: how much time was spent pulling cases from storage one at a time).